File: Data Structures & Algorithms/daily-temperatures/submission-17.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        
        stack = []
        tempDifferences = [0] * len(temperatures)

        for r in range(len(temperatures)):
            while stack and temperatures[r] > temperatures[stack[-1]]:
                l = stack.pop()
                tempDifferences[l] = r - l
            stack.append(r)
        
        return tempDifferences

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-17.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        std::stack<int> s;
        for (auto& token : tokens) {
            if (token == "+" || token == "-" || token == "*" || token == "/") {
                int digitTwo = s.top(); s.pop();
                int digitOne = s.top(); s.pop();
                if (token == "+") s.push(digitOne + digitTwo);
                else if (token == "-") s.push(digitOne - digitTwo);
                else if (token == "*") s.push(digitOne * digitTwo);
                else if (token == "/") s.push(digitOne / digitTwo); // truncates toward 0, matches C++ int division
            } else {
                s.push(std::stoi(token));
            }
        }
        return s.top();
    }
};
